lib/client.js

```diff
@@ -116,11 +117,33 @@ var makeRequest = function(client, method, url, payload) {
  *   baseUrl:         'http://.../v1'   // baseUrl for API requests
  *   exchangePrefix:  'queue/v1/'       // exchangePrefix prefix
  *   retries:         5,                // Maximum number of retries
+ *   stats:           function(obj) {}  // Callback for reporting statistics
  * }
  *
  * `baseUrl` and `exchangePrefix` defaults to values from reference.
+ *
+ * `options.stats` is an optional function that takes an object with the
+ * following properties:
+ * ```js
+ * {
+ *   duration:     0,              // API call time (ms) including all retries
+ *   retries:      0,              // Number of retries
+ *   method:       'createTask',   // Name of method called
+ *   success:      0 || 1,         // Success or error
+ *   resolution:   201,            // Status code
```

Comment:
Would this be the status code or something like 'http-ok' ? 

Based on: https://github.com/taskcluster/taskcluster-client/pull/20/files#diff-50cfa59973c04321b5da0c6da0fdf4feR289

it looks like the resolution is prefixed with 'http-' and I think res.status is something like 'ok'